lib/src/yaml_node.dart

```diff
@@ -26,9 +26,9 @@ abstract class YamlNode {
   ///
   /// [SourceSpan.message] can be used to produce a human-friendly message about
   /// this node.
-  SourceSpan get span => _span;
+  SourceSpan? get span => _span;
```

Comment:
It's kind of hard to imagine that a node might not have a span. It would be good to understand whether that's intentional, in which case the comment should be updated to explain when it will be `null`, or accidental, in which case the field should be made non-nullable and other code fixes as appropriate.

Reply:
@munificent: thoughts here?

Reply:
This pattern also shows up in `package:html`. In that package there is a setting which enables or disables tracking source spans (I assume as a performance optimization?) https://github.com/dart-lang/html/pull/139#discussion_r507694093

Are these also sometimes tracked and sometimes not tracked?

Reply:
Thanks @natebosch!  I don't see that pattern being used here.  Unless @nex3 or @munificent can think of any reason that span should remain nullable, I'm included to go w/ Brian's suggestion and mark `_span` as late and make it not nullable.

Reply:
In some packages, we use a `null` source span to indicate a node that's constructed in-memory without any source information, but it looks like this package uses a custom `NullSourceSpan` class for that instead so this should be safely non-nullable.

Reply:
What do you think about adding a constructor so it doesn't need to be `late`?

https://github.com/dart-lang/yaml/pull/95

Reply:
Great idea @natebosch.